CMakeLists.txt

```diff
@@ -440,9 +440,9 @@ elseif (APPLE)
   set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-rpath,@loader_path -Wl,-rpath,@loader_path/..")
 else()
   # LINUX
-  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath,$ORIGIN")
-  set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,-rpath,$ORIGIN -Wl,-rpath,$ORIGIN/..")
-  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -Wl,-rpath,$ORIGIN -Wl,-rpath,$ORIGIN/..")
+  set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-rpath,\\$ORIGIN")
```

Comment:
Just curious, is ORIGIN a shell variable or a cmake variable? If it's a cmake variable, you may need the variable interpolation to happen here.

Reply:
It is a cmake variable. I'm not sure I understand what you mean by variable interpolation here. Are you suggesting it should be `\\${ORIGIN}` rather than just `\\$ORIGIN`?

Reply:
I was wrong. The literal sting `$ORIGIN`needs to be passed to the linker. It should not be expanded. It should not be `${ORIGIN}`.

Reply:
Sorry, I should make myself clear. variable interpolation means expansion.

Reply:
https://stackoverflow.com/questions/29584348/escaping-dollar-sign-in-cmake.

Reply:
shell will receive "\\$" as well, where reference variable after `$` will be expanded as well, with a pending `\`.

a single back slach is enough for shell.
```
echo "\$HOST"
```

it should also work for CMake.